eclair-core/src/main/scala/fr/acinq/eclair/blockchain/electrum/ElectrumClientPool.scala

```diff
@@ -73,7 +73,7 @@ class ElectrumClientPool(serverAddresses: Set[InetSocketAddress])(implicit val e
 
     case Event(ElectrumClient.AddStatusListener(listener), d: ConnectedData) =>
       statusListeners += listener
-      listener ! ElectrumClient.ElectrumReady(d.tips(d.master))
+      listener ! ElectrumClient.ElectrumReady(d.tips(d.master), addresses(d.master))
```

Comment:
I'm not sure about these `addresses(d.master)`, which may fail if key is not present. Maybe add a `if addresses.contains(sender)` like in others cases?

Reply:
You mean `addresses.contains(d.master)` ?

Reply:
Yes, something like that which would be at the `case .... =>` level.

I figure this could be better done via smarter typing but I don't have a proposition readily available.